projects/transformers/run_utils.py

```diff
@@ -610,6 +610,7 @@ def init_tokenizer(model_args):
         use_fast=model_args.use_fast_tokenizer,
         revision=model_args.model_revision,
         use_auth_token=True if model_args.use_auth_token else None,
+        config=config,
```

Comment:
This is interesting. How is the tokenizer affected by the changes made to sparsify the model? Shouldn't we use the same tokenizer used for both cases?

If we change the tokenizer, the caching tokenized dataset process will silently fail, since today is only based on dataset+dataset_config and assumes there will be no changes to the tokenizer (for example if we change the number of tokens in the model embedding layer). Keep in mind tokenization is very costly and it will take almost 2 days to re-tokenize wikipedia, hence the forced loading from cache style. If we move to larger datasets that is going to be even worse. 

Not requiring immediate changes, but just want to raise awareness about this issue since there is potential to introduce bugs if we are passing this config to the tokenizer. We should think about changing the behavior in run.py and accept two different model arguments, one to the tokenizer which will be fixed to `bert-based-case` unless specifically defined otherwise, and the other to the model.

Reply:
Good point. I'm not sure why exactly I added it. Possibly just because the Tokenizer can accept the config. I just ran it without this line, and no errors are introduced. So I think I'll remove it just to be safe.